python/ray/serve/experimental/gradio_visualize_graph.py

```diff
@@ -39,6 +41,74 @@ def lazy_import_gradio():
     return _gradio
 
 
+def get_block(node, **kwargs):
+    gr = lazy_import_gradio()
+
+    result_type = node.get_result_type()
+    node_name = _DAGNodeNameGenerator().get_node_name(node)
+    if result_type is None:
+        logger.warning(
+            f'Return type for "{node_name}" node was not provided. '
+            "Defaulting to gr.Textbox."
+        )
+        return gr.Textbox(**kwargs)
+
+    # result_type is generated by ray.experimental.gradio_utils.type_to_string()
+    if result_type == "int":
+        return gr.Number(precision=0, **kwargs)
+    elif result_type == "float":
+        return gr.Number(**kwargs)
+    elif result_type == "str":
+        return gr.Textbox(**kwargs)
+    elif result_type == "bool":
+        return gr.Checkbox(**kwargs)
+    elif result_type == "pandas.core.frame.DataFrame":
+        return gr.Dataframe(**kwargs)
+    elif (
+        result_type == "list"
+        or result_type == "dict"
+        or result_type == "typing.List"
+        or result_type == "typing.Dict"
+        or result_type == "numpy.ndarray"
+    ):
+        return gr.JSON(**kwargs)
+    elif result_type == "torch.tensor":
+        return gr.Image(**kwargs)
+    else:
+        from PIL import ImageFile
+
+        if issubclass(locate(result_type), ImageFile.ImageFile):
+            return gr.Image(**kwargs)
+
+    logger.warning(
+        f"Return type for {node_name} node is not valid. Defaulting to gr.Textbox."
+    )
+    return gr.Textbox(**kwargs)
+
+
+def postprocessing(data):
```

Comment:
Could you provide more details about this function in the docstring? Specifically, what does it mean when a type is `not supported by Gradio`?